test_conformance/extensions/cl_khr_semaphore/test_semaphores_cross_queue.cpp

```diff
@@ -0,0 +1,321 @@
+//
+// Copyright (c) 2024 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include <vector>
+
+#include "semaphore_base.h"
+
+namespace {
+
+int num_elems = 0;
```

Comment:
Can we get rid of this non-const global variable?  We should pass the number of events to the test class constructor and store it per-class, instead.

Reply:
Done! Side note: as consequence I had to correct other tests due to the fact all uses the same interface.

Reply:
OK, I see.  As an alternative, if you aren't comfortable changing so many existing tests, consider adding overloads for `MakeAndRunTest`, so there is one version with an `nelems` parameter and one version without.

Reply:
Indeed, it crossed my mind. After all, the work has already been done, I think it is a bit more consistent with other `MakeAndRun` procedures.